### DISCUSSION TITLE
force rust-analyzer to expand macro_rules?

### DISCUSSION THREAD

**Discussion by evbo** [rust-lang/rust-analyzer#17971] [Q&A]
For what ever reason I need to manually "restart" rust-analyzer if I change a single line of code or else it thinks `const` and `static` variables no longer exist generated by this macro_rules:

```rust
macro_rules! make_types {
    ( $l:ident, ($($n:ident $c:literal $d:expr),*)) => {
        $( pub const $n: &str = $c; )*

        lazy_static! {
            pub static ref $l: Vec<(&'static str, DataType)> = vec![
                $( ($n, $d), )*
            ];
        }
    };
}
```

it also strangely highlights this `use` as "unresolved import" until I restart after every code change: `use lazy_static::lazy_static;`

How can I hint rust-analyzer to expand macros after lines of code change so I don't see as much red in my editor?

**Comment by evbo** [comment]
just found this open issue, maybe related?: https://github.com/rust-lang/rust-analyzer/issues/17837

**Reply by flodiebold** [reply]
Doesn't look likely.

**Comment by flodiebold** [comment]
> How can I hint rust-analyzer to expand macros after lines of code change so I don't see as much red in my editor?

You can't, because you shouldn't need to. Can you reproduce the problem in a new project?
